scipy/stats/_stats_py.py

```diff
@@ -6938,23 +6950,35 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
     Ttest_indResult(statistic=3.4463884028073513,
                     pvalue=0.01369338726499547)
     """
+    xp = array_namespace(a, b)
+
+    default_float = xp.asarray(1.).dtype
+    if xp.isdtype(a.dtype, 'integral'):
+        a = xp.astype(a, default_float)
```

Comment:
I think I'm waiting for consensus one what should be done in this situation before factoring out this code. Ultimately it could go in the `_axis_nan_policy` decorator or something.